pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -134,44 +134,54 @@ func (s *FlinkStateMachine) Handle(ctx context.Context, application *v1alpha1.Fl
 }
 
 func (s *FlinkStateMachine) handle(ctx context.Context, application *v1alpha1.FlinkApplication) error {
+	var appErr error
+	updateK8Object := !shouldUpdateK8Object
```

Comment:
You do not need `shouldUpdateK8Object`. Why not just initialize everything to false ?

Reply:
Also you can simply make handle() return bool, error, and pass it to caller func `Handle` (notice caps)